visualization/tools/pcd_viewer.cpp

```diff
@@ -214,224 +214,65 @@ pp_callback (const pcl::visualization::PointPickingEvent& event, void* cookie)
-  float normals_scale = NORMALS_SCALE;
-  pcl::console::parse_argument (argc, argv, "-normals_scale", normals_scale);
-
-  int pc = 0;
-  pcl::console::parse_argument (argc, argv, "-pc", pc);
-  float pc_scale = PC_SCALE;
-  pcl::console::parse_argument (argc, argv, "-pc_scale", pc_scale);
-
-  bool use_vbos = false;
-  pcl::console::parse_argument (argc, argv, "-vbo_rendering", use_vbos);
-  if (use_vbos) 
-    print_highlight ("Vertex Buffer Object (VBO) visualization enabled.\n");
-
-  bool use_pp   = pcl::console::find_switch (argc, argv, "-use_point_picking");
-  if (use_pp) 
-    print_highlight ("Point picking enabled.\n");
-
-  bool use_optimal_l_colors = pcl::console::find_switch (argc, argv, "-optimal_label_colors");
-  if (use_optimal_l_colors)
-    print_highlight ("Optimal glasbey colors are being assigned to existing labels.\nNote: No static mapping between label ids and colors\n");
-
-  // If VBOs are not enabled, then try to use immediate rendering
-  bool use_immediate_rendering = false;
-  if (!use_vbos)
-  {
-    pcl::console::parse_argument (argc, argv, "-immediate_rendering", use_immediate_rendering);
-    if (use_immediate_rendering) 
-      print_highlight ("Using immediate mode rendering.\n");
-  }
-
-  // Multiview enabled?
-  int y_s = 0, x_s = 0;
-  double x_step = 0, y_step = 0;
-  if (mview)
-  {
-    print_highlight ("Multi-viewport rendering enabled.\n");
-
-    y_s = static_cast<int>(floor (sqrt (static_cast<float>(p_file_indices.size () + vtk_file_indices.size ()))));
-    x_s = y_s + static_cast<int>(ceil (double (p_file_indices.size () + vtk_file_indices.size ()) / double (y_s) - y_s));
-
-    if (p_file_indices.size () != 0)
-    {
-      print_highlight ("Preparing to load "); print_value ("%d", p_file_indices.size ()); print_info (" pcd files.\n");
-    }
-
-    if (vtk_file_indices.size () != 0)
-    {
-      print_highlight ("Preparing to load "); print_value ("%d", vtk_file_indices.size ()); print_info (" vtk files.\n");
-    }
-
-    x_step = static_cast<double>(1.0 / static_cast<double>(x_s));
-    y_step = static_cast<double>(1.0 / static_cast<double>(y_s));
-    print_value ("%d", x_s);    print_info ("x"); print_value ("%d", y_s);
-    print_info (" / ");      print_value ("%f", x_step); print_info ("x"); print_value ("%f", y_step);
-    print_info (")\n");
-  }
-
-  // Fix invalid multiple arguments
-  if (psize.size () != p_file_indices.size () && psize.size () > 0)
-    for (size_t i = psize.size (); i < p_file_indices.size (); ++i)
-      psize.push_back (1);
-  if (opaque.size () != p_file_indices.size () && opaque.size () > 0)
-    for (size_t i = opaque.size (); i < p_file_indices.size (); ++i)
-      opaque.push_back (1.0);
-
-  if (shadings.size () != p_file_indices.size () && shadings.size () > 0)
-    for (size_t i = shadings.size (); i < p_file_indices.size (); ++i)
-      shadings.push_back ("flat");
-
-  // Create the PCLVisualizer object
+handlePointCloud (int & argc, char** argv
```

Comment:
I haven't reviewed this but I can tell you already I'm not gonna accept this function signature.

Reply:
I agree, this prototype should be changed. Always check what I propose, I still learn from you feedback.